packages/flutter/test/material/data_table_test.dart

```diff
@@ -918,4 +919,115 @@ void main() {
+      expect(nameText, findsOneWidget);
+      final Finder nameCell = find.ancestor(of: find.text('Name'), matching: find.byType(Container)).first;
+      expect(tester.getCenter(nameText), equals(tester.getCenter(nameCell)));
+      expect(find.descendant(of: nameCell, matching: find.byType(Icon)), findsNothing);
+    }
+
+    await tester.pumpWidget(MaterialApp(
+      home: Material(child: buildTable(
+        sortEnabled: true,
+      )),
+    ));
+
+    {
+      final Finder nameText = find.text('Name');
+      expect(nameText, findsOneWidget);
+      final Finder nameCell = find.ancestor(of: find.text('Name'), matching: find.byType(Container)).first;
+      expect(find.descendant(of: nameCell, matching: find.byType(Icon)), findsOneWidget);
+    }
+
+    await tester.pumpWidget(MaterialApp(
+      home: Material(child: buildTable(
+        sortEnabled: false,
+      )),
+    ));
+
+    {
+      final Finder nameText = find.text('Name');
+      expect(nameText, findsOneWidget);
+      final Finder nameCell = find.ancestor(of: find.text('Name'), matching: find.byType(Container)).first;
+      expect(tester.getCenter(nameText), equals(tester.getCenter(nameCell)));
+      expect(find.descendant(of: nameCell, matching: find.byType(Icon)), findsNothing);
+    }
+  });
+
+  testWidgets('DataTable correctly renders with a mouse', (WidgetTester tester) async {
+    // Regression test for a bug described in
+    // https://github.com/flutter/flutter/pull/43735#issuecomment-589459947
+    // Filed at https://github.com/flutter/flutter/issues/51152
+    Widget buildTable({ int sortColumnIndex }) {
+      return DataTable(
+        sortColumnIndex: sortColumnIndex,
+        columns: <DataColumn>[
+          const DataColumn(
+            label: Expanded(child: Center(child: Text('column1'))),
+            tooltip: 'Column1',
+          ),
+          DataColumn(
+            label: const Expanded(child: Center(child: Text('column2'))),
+            tooltip: 'Column2',
+            onSort: (_, __) {},
+          ),
+        ],
+        rows: const <DataRow>[
+          DataRow(
+            cells: <DataCell>[
+              DataCell(Text('Content1')),
+              DataCell(Text('Content2')),
+            ],
+          ),
+        ]
+      );
+    }
+
+    await tester.pumpWidget(MaterialApp(
+      home: Material(child: buildTable()),
+    ));
+
+    expect(tester.renderObject(find.text('column1')).attached, true);
+    expect(tester.renderObject(find.text('column2')).attached, true);
+
+    final TestGesture gesture = await tester.createGesture(kind: PointerDeviceKind.mouse);
+    await gesture.addPointer(location: Offset.zero);
+    addTearDown(gesture.removePointer);
+
+    await tester.pumpAndSettle();
+    expect(tester.renderObject(find.text('column1')).attached, true);
+    expect(tester.renderObject(find.text('column2')).attached, true);
+
+    // Wait for the tooltip timer
+    await tester.pumpAndSettle(const Duration(seconds: 1));
```

Comment:
Do you want to check something after this `pumpAndSettle`?

Reply:
Not really. This `pumpAndSettle` is mostly to prevent an exception thrown by `Tooltip` when its timer expires and schedules a new frame when the view has been destroyed. I've updated the comment to make it clearer.